src/modals/examples/modal-overlay.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.modals.componenet:pfModalOverlay
+ * @element pfModalOverlay
+ * @restrict E
+ *
+ * @description
+ * The Modal Overlay pattern provides a way to quickly load and display important information to the user without navigating
+ * away from the current page. It utilizes the {@link https://angular-ui.github.io/bootstrap/#modal Angular UI modal}
+ * and Patternfly styling. Use cases for Modal Overlays can vary a great deal, but some examples include the following:
+ * <ul>
+ * <li>To remind or prompt users
+ * <li>To load dialogs that require user input before advancing
+ * <li>To load tasks which require a user’s full attention, such as stepping through a wizard flow
+ * <li>To present important information or warnings
+ * </ul>
+ *
+ * @param {boolean} isOpen Flag indicating that the modal should be opened
+ * @param {function} onClose Function to call when the modal is closed
+ * @param {string} modalId Id of the modal
+ * @param {string} modalBodyPath The path to the html template to populate the modal body
+ * @param {string} modalTitle The title of the modal displayed in the header
+ * @param {string=} titleId Id of the title. "modalTitle" by default
+ * @param {boolean=} hideCloseIcon Flag indicating that the modal should hide the 'x' close icon.
+ * @param {boolean=} backdropClose Flag indicating that the modal should close if user clicks background. False by default
+ * @param {array} actionButtons array of button objects. Each button can have the following properties:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.label        - the text to display on the button
+ * <li>.class        - (optional) classes to add to the button
+ * <li>.actionFn     - (optional) use defined function to call when the button is clicked
+ * <li>.isDisabled   - (optional) boolean true if the button should be disabled by default
+ * <li>.isCancel     - (optional) boolean true is the button should cancel and dismiss the modal
+ * </ul>
+ *
+ * @example
+ <example module="patternfly.modals">
+
+ <file name="index.html">
+ <div ng-controller="DemoModalOverlayCtrl">
+    <button ng-click="open()" class="btn btn-default">Launch Modal Overlay</button>
+    <pf-modal-overlay is-open="isOpen"
+          on-close="onClose()"
+          modal-id="modalId"
+          modal-body-path="modalBodyPath"
```

Comment:
Hi, is it possible to use angular's `transclusion` here?  See [pfFilterPanel](https://www.patternfly.org/angular-patternfly/#/api/patternfly.filters.component:pfFilterPanel) or [pfListView](https://www.patternfly.org/angular-patternfly/#/api/patternfly.views.component:pfListView) for examples on how we use transclusion.    [ng-transclude](https://github.com/patternfly/angular-patternfly/blob/master/src/filters/filter-panel/filter-panel.html#L7) in filter-panel.html.